File: stack.js
class Stack {
  constructor(maxSize) {
    this.maxSize = maxSize;
    this.stack = [];
    this.top = -1;
  }

  push(value) {
    if (this.top >= this.maxSize - 1) {
      console.log("Stack overflow. Cannot push " + value + " to the stack.");
      return;
    }
    this.stack[++this.top] = value;
    console.log("Pushed " + value + " to the stack.");
  }

  pop() {
    if (this.top < 0) {
      console.log("Stack underflow. No element to pop.");
      return -1;
    }
    const value = this.stack[this.top--];
    console.log("Popped " + value + " from the stack.");
    return value;
  }
}

const stack = new Stack(5);
stack.push(10);
stack.push(20);
stack.push(30);
stack.push(40);
stack.push(50);

stack.pop();
stack.pop();
stack.pop();
stack.pop();
stack.pop();

File: stack.c
#include <stdio.h>

#define MAX_SIZE 5

int stack[MAX_SIZE];
int top = -1;

void push(int value) {
    if (top >= MAX_SIZE - 1) {
        printf("Stack overflow. Cannot push %d to the stack.\n", value);
        return;
    }
    stack[++top] = value;
    printf("Pushed %d to the stack.\n", value);
}

int pop() {
    if (top < 0) {
        printf("Stack underflow. No element to pop.\n");
        return -1;
    }
    int value = stack[top--];
    printf("Popped %d from the stack.\n", value);
    return value;
}

int main() {
    push(10);
    push(20);
    push(30);
    push(40);
    push(50);
   

    pop();
    pop();
    pop();
    pop();
    pop();
    

    return 0;
}